lib/ui/surveys/survey_view.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:survey_flutter_ic/gen/assets.gen.dart';
+import 'package:survey_flutter_ic/model/survey_model.dart';
+import 'package:survey_flutter_ic/widget/circle_button.dart';
+
+class SurveyView extends ConsumerStatefulWidget {
+  final List<SurveyModel> surveys;
+  final Function(int) onPageChanged;
+  final Function(SurveyModel) onSurveySelected;
+
+  const SurveyView(
+      {Key? key,
+      required this.surveys,
+      required this.onPageChanged,
+      required this.onSurveySelected})
+      : super(key: key);
+
+  @override
+  ConsumerState<SurveyView> createState() => _SurveyViewState();
+}
+
+class _SurveyViewState extends ConsumerState<SurveyView> {
+  final PageController _pageController = PageController(initialPage: 0);
+
+  @override
+  void dispose() {
+    super.dispose();
+    _pageController.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
@hoangnguyen92dn This function is too long, please help to split it out to many parts

Reply:
@doannimble There are 3 single widgets to display Title, Description and CircleButton. So I think it's simple enough in the `build` function 😄

Reply:
@hoangnguyen92dn We do not care how complex it is, when reading the long function, it is hard for us to readability. Especially for the maintainer. And we should not make the main build function too long
FYI: We have a lint rule in Varun project: the LOC should not be over 30 🙏